File: BINDINGS/cxx/eflxx_examples/src/elementaryxx/full/test_toggle.cpp
#include "test.h"

void test_toggle (void *data, Evas_Object *obj, void *event_info)
{
  Window *win = Window::factory ("toggles", ELM_WIN_BASIC);
  win->setTitle ("Toggles");
  win->setAutoDel (true);

  Background *bg = Background::factory (*win);
  win->addObjectResize (*bg);
  bg->setWeightHintSize (EVAS_HINT_EXPAND, EVAS_HINT_EXPAND);
  bg->show ();

  Box *bx = Box::factory (*win);
  win->addObjectResize (*bx);
  bx->setWeightHintSize (EVAS_HINT_EXPAND, EVAS_HINT_EXPAND);
  bx->show ();

  Icon *ic = Icon::factory (*win);
  ic->setFile (searchPixmapFile ("elementaryxx/logo_small.png"));
  ic->setAspectHintSize (EVAS_ASPECT_CONTROL_VERTICAL, Size (1, 1));
  Toggle *tg = Toggle::factory (*win);
  tg->setWeightHintSize (EVAS_HINT_EXPAND, EVAS_HINT_EXPAND);
  tg->setAlignHintSize (EVAS_HINT_FILL, 0.5);
  tg->setLabel ("Icon sized to toggle");
  tg->setIcon (*ic);
  tg->setState (true);
  tg->setLabelsStates ("Yes", "No");
  bx->packEnd (*tg);
  tg->show ();
  ic->show ();

  Icon *ic2 = Icon::factory (*win);
  ic2->setFile (searchPixmapFile ("elementaryxx/logo_small.png"));
  ic2->setScale (false, false);
  Toggle *tg2 = Toggle::factory (*win);
  tg2->setLabel ("Icon no scale");
  tg2->setIcon (*ic2);
  bx->packEnd (*tg2);
  tg2->show ();
  ic2->show ();

  Toggle *tg3 = Toggle::factory (*win);
  tg3->setLabel ("Label Only");
  tg3->setLabelsStates ("Big long fun times label", "Small long happy fun label");
  bx->packEnd (*tg3);
  tg3->show ();

  Icon *ic3 = Icon::factory (*win);
  ic3->setFile (searchPixmapFile ("elementaryxx/logo_small.png"));
  ic3->setScale (false, false);
  Toggle *tg4 = Toggle::factory (*win);
  tg4->setIcon (*ic3);
  bx->packEnd (*tg4);
  tg4->show ();
  ic3->show ();

  win->show ();
}

File: BINDINGS/python/python-elementary/tests/test_toggle.py
#!/usr/bin/env python
import os
import elementary
import edje
import ecore
import evas

#----- Toggles -{{{-
def toggles_clicked(obj, it):
    win = elementary.Window("toggles", elementary.ELM_WIN_BASIC)
    win.title_set("Toggles")
    win.autodel_set(True)

    bg = elementary.Background(win)
    win.resize_object_add(bg)
    bg.size_hint_weight_set(evas.EVAS_HINT_EXPAND, evas.EVAS_HINT_EXPAND)
    bg.show()

    bx = elementary.Box(win)
    win.resize_object_add(bx)
    bx.size_hint_weight_set(evas.EVAS_HINT_EXPAND, evas.EVAS_HINT_EXPAND)
    bx.show()

    ic = elementary.Icon(win)
    ic.file_set("images/logo_small.png")
    ic.size_hint_aspect_set(evas.EVAS_ASPECT_CONTROL_VERTICAL, 1, 1)
    tg = elementary.Toggle(win)
    tg.label_set("Icon sized to toggle")
    tg.icon_set(ic)
    tg.state_set(True)
    tg.states_labels_set("Yes", "No")
    bx.pack_end(tg)
    tg.show()
    ic.show()

    ic = elementary.Icon(win)
    ic.file_set("images/logo_small.png")
    ic.scale_set(0, 0)
    tg = elementary.Toggle(win)
    tg.label_set("Icon no scale")
    tg.icon_set(ic)
    bx.pack_end(tg)
    tg.show()
    ic.show()

    tg = elementary.Toggle(win)
    tg.label_set("Label Only")
    tg.states_labels_set("Big", "Small")
    bx.pack_end(tg)
    tg.show()

    ic = elementary.Icon(win)
    ic.file_set("images/logo_small.png")
    ic.scale_set(0, 0)
    tg = elementary.Toggle(win)
    tg.icon_set(ic)
    bx.pack_end(tg)
    tg.show()
    ic.show()

    win.show()
# }}}

#----- Main -{{{-
if __name__ == "__main__":
    elementary.init()

    toggles_clicked(None, None)

    elementary.run()
    elementary.shutdown()
# }}}
# vim:foldmethod=marker
